nsfc-plugin/src/test/java/org/osc/controller/nsfc/OSGiIntegrationTest.java

```diff
@@ -251,234 +115,8 @@ public SSLContext getSslContext() {
-        assertNull(this.osClient.sfc().flowclassifiers().get(hookId));
-
-        inspectedPortCheck = this.osClient.networking().port().get(INSPECTED_ID);
-        assertNotNull(inspectedPortCheck);
-        assertFalse(inspectedPortCheck.getProfile() != null
-                           && inspectedPortCheck.getProfile().get(KEY_HOOK_ID) != null);
-
-        portChainCheck = this.osClient.sfc().portchains().get(sfcId);
-        assertFalse(portChainCheck.getFlowClassifiers() != null
-                          && portChainCheck.getFlowClassifiers().contains(hookId));
-
-        // TEST CALL
-        this.redirApi.deleteNetworkElement(sfc);
-        assertNull(this.osClient.sfc().portchains().get(sfc.getElementId()));
-    }
-
-    private void cleanAllOnOpenstack() {
-        List<? extends PortChain> portChains = this.osClient.sfc().portchains().list();
-        List<? extends FlowClassifier> flowClassifiers = this.osClient.sfc().flowclassifiers().list();
-        List<? extends PortPairGroup> portPairGroups = this.osClient.sfc().portpairgroups().list();
-        List<? extends PortPair> portPairs = this.osClient.sfc().portpairs().list();
-
-        for (PortChain pc : portChains) {
-            this.osClient.sfc().portchains().delete(pc.getId());
-        }
-        for (FlowClassifier fc : flowClassifiers) {
-            this.osClient.sfc().flowclassifiers().delete(fc.getId());
-        }
-        for (PortPairGroup ppg : portPairGroups) {
-            this.osClient.sfc().portpairgroups().delete(ppg.getId());
-        }
-        for (PortPair pp : portPairs) {
-            this.osClient.sfc().portpairs().delete(pp.getId());
-        }
-
-        assertEquals("Failed clean port chains!", 0, this.osClient.sfc().portchains().list().size());
-        assertEquals("Failed clean flow classifiers!", 0, this.osClient.sfc().flowclassifiers().list().size());
-        assertEquals("Failed clean port pair groups!", 0, this.osClient.sfc().portpairgroups().list().size());
-        assertEquals("Failed clean port pairs!", 0, this.osClient.sfc().portpairs().list().size());
-    }
-
-    private void assertIngressEgressOk(PortPairElement inspectionPortElement) {
-        PortPair portPairCheck = this.osClient.sfc().portpairs().get(inspectionPortElement.getElementId());
-        assertEquals(inspectionPortElement.getEgressPort().getElementId(), portPairCheck.getEgressId());
-        assertEquals(inspectionPortElement.getEgressPort().getElementId(), portPairCheck.getEgressId());
-    }
-
-    private void assertPortPairGroupIsOk(PortPairElement inspectionPortElement) {
-        assertNotNull(inspectionPortElement.getPortPairGroup());
-        PortPairGroup ppgCheck = this.osClient.sfc().portpairgroups().get(inspectionPortElement.getParentId());
-        assertNotNull(ppgCheck);
-        assertNotNull(ppgCheck.getPortPairs());
-        assertTrue(ppgCheck.getPortPairs().contains(inspectionPortElement.getElementId()));
+       assertNotNull(this.api);
```

Comment:
does this fail when unresolved osgi libraries exist?

Reply:
Some potential class-not-found exceptions are not detected, but only runtime ones.